enterprise/frontend/src/embedding-sdk/cli/utils/get-component-snippets.ts

```diff
@@ -26,33 +35,53 @@ export function getComponentSnippets(options: Options) {
     ? ANALYTICS_PROVIDER_SNIPPET_WITH_USER_SWITCHER
     : ANALYTICS_PROVIDER_SNIPPET_MINIMAL;
 
-  const components = [
+  const components: SampleComponent[] = [
     {
-      name: "analytics-provider",
-      content: analyticsProviderSnippet.trim(),
+      fileName: "analytics-provider",
+      componentName: "AnalyticsProvider",
+      content: withNextJsUseClientDirective(
+        analyticsProviderSnippet,
+        isNextJs,
+      ).trim(),
     },
     {
-      name: "embedding-provider",
-      content: embeddingProviderSnippet.trim(),
+      fileName: "embedding-provider",
+      componentName: "EmbeddingProvider",
+      content: withNextJsUseClientDirective(
+        embeddingProviderSnippet,
+        isNextJs,
+      ).trim(),
```

Comment:
This is for the `useContext` that gives us access to the light/dark mode key, so we can provide the correct theme to MetabaseProvider